core/src/main/scala/org/apache/spark/storage/BlockManager.scala

```diff
@@ -428,77 +458,46 @@ private[spark] class BlockManager(
       Option(
         shuffleBlockResolver.getBlockData(blockId.asInstanceOf[ShuffleBlockId]).nioByteBuffer())
     } else {
-      doGetLocal(blockId, asBlockResult = false).asInstanceOf[Option[ByteBuffer]]
-    }
-  }
-
-  private def doGetLocal(blockId: BlockId, asBlockResult: Boolean): Option[Any] = {
-    blockInfoManager.lockForReading(blockId) match {
-      case None =>
-        logDebug(s"Block $blockId was not found")
-        None
-      case Some(info) =>
-        doGetLocal(blockId, info, asBlockResult)
+      blockInfoManager.lockForReading(blockId) match {
```

Comment:
need to clarify in javadocs the locking semantics of `getLocalValues` and `getLocalBytes`. In particular, both of these methods acquire the read lock. In the iterator case we release the lock when the iterator is drained. In the bytes case the caller is responsible for releasing the locks. In both cases, if we couldn't get the blocks for some reason, we release the lock and throw an exception instead. (Is that right?)